### PR TITLE
DROPDOWN Add a dropdown to molecules

### PR DESCRIPTION
**Pull request by mariadoesmath**
Add a dropdown to molecules based off of the [design](https://www.figma.com/file/AV6iiYaGYNOk5draCeHCMm/Offer-Letter?node-id=88%3A39409) of the dropdown added for offer letters

See a real example in the generated storybook page for this PR [here](https://deploy-preview-39--hireology-storybook.netlify.app/?path=/story/dropdown--default)

### CODE REVIEW

File: src/components/Dropdown/Dropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    actions: [],
+  };
+
+  state = {
+    activeSuggestionIndex: -1,
+    selectedText: null,
+  };
+
+  /**
+   * 38: up arrow - move to the previous suggestion
```
Comment (mariadoesmath) [inline src/components/Dropdown/Dropdown.jsx:38]: This was copy pastad from TypeAheadSearch, let's update
- [ ] "suggestion" => "option"

**Comment by mariadoesmath** [comment]
After talking with Jess, it seems we also need to have the ability to have an item selected by default

- [ ] Add ability to select item by default

**Comment by mdchristians** [comment]
Should this be using a select element?

**Comment by mariadoesmath** [comment]
> Should this be using a select element?

Idk, I went back and forth on that. I ended up saying no because there are use cases (like for [offer letter](https://www.figma.com/file/AV6iiYaGYNOk5draCeHCMm/Offer-Letter?node-id=88%3A39409))where we are going to have an item in the dropdown that you don't _actually_ select on click, so I thought it'd be better to let each option have it's own individual callback - which i don't believe you can do with a select. It's also really hard to stylize a select across all browsers and on mobile devices because they all add their own individual styles that look like trash. I did make the above dropdown keyboard friendly so it's still accessible - though I'm not 100% sure how it'd stand up to a screen reader.

**Comment by mdchristians** [comment]
> Idk, I went back and forth on that. I ended up saying no because there are use cases (like for offer letter) where we are going to have an item in the dropdown that you don't actually select on click, so I thought it'd be better to let each option have it's own individual callback - which i don't believe you can do with a select. It's also really hard to stylize a select across all browsers and on mobile devices because they all add their own individual styles that look like trash. I did make the above dropdown keyboard friendly so it's still accessible - though I'm not 100% sure how it'd stand up to a screen reader.

Okay, that makes sense. In this case it might be a good option to use a render prop and export optional base components as well to make it more flexible. You may also want to take a look at [React Select @ v1](https://v1.react-select.com/) and using that, as it should be compatible with the app.

**Pull request merged**